lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1102,6 +1102,31 @@ struct FIRStmtParser : public FIRScopedParser {
 
   // Extra information maintained across a module.
   FIRModuleContext &moduleContext;
+
+public:
+  /// Utility to get the CircuitTarget
+  std::string getCircuitTarget() {
+    return "~" + builder.getBlock()
+                     ->getParentOp()
```

Comment:
I don't think the `->getParentOp()` is needed, given the getParentOfType, but I'd just suggest adding the `circuit` as an instance variable to the  parser state thing.  That will make this more efficient and simpler.

Reply:
Will be fixed in the rebase. I change this to statefully, globally store:
```c++
StringRef circuitTarget;
StringRef moduleTarget;
```
This utility is then:
```c++
Twine getReferenceTarget(Twine ref);
```